daffodil-cli/src/it/scala/org/apache/daffodil/schematron/TestEmbedded.scala

```diff
@@ -0,0 +1,105 @@
+  }
+
+  @Test def unitPriceWithoutValidation(): Unit = withShell() {
+    val data = mktmp("widget,monday,1,$5.00,$5.00")
+    val schema = "xsd/unit_price.dfdl.xsd"
+    s"parse -r list -s {{$schema}} $data" -> contains("</ex:list>\n")
+  }
+
+  @Test def unitPriceWithValidation(): Unit = withShell(FailureErrorCodeResult, true) {
+    val data = mktmp("widget,monday,1,$5.00,$6.00")
+    val schema = "xsd/unit_price.dfdl.xsd"
+    s"parse -r list --validate schematron={{$schema}} -s {{$schema}} $data" -> sequence(
+      contains("</ex:list>\n"),
+      anyLines(3)
+    )
+  }
+
+  @Test def unitPriceWithValidationCheckMessage(): Unit = withShell(FailureErrorCodeResult, true) {
+    val data = mktmp("widget,monday,5,$5.00,$25.00||gadget,tuesday,1,$10.00,$11.00")
+    val schema = "xsd/unit_price.dfdl.xsd"
+    s"parse -r list --validate schematron={{$schema}} -s {{$schema}} $data" -> sequence(
+      contains("</ex:list>\n"),
+      contains("[error] Validation Error: wrong unit price for gadget, tuesday\n"),
+      anyLines(2)
+    )
+  }
+
+  @Test def extends1(): Unit = withShell() {
+    val data = mktmp("bob;l;smith")
+    val schema = "xsd/extends-1.dfdl.xsd"
+    s"parse --validate schematron={{$schema}} -s {{$schema}} $data" -> contains("</name>\n")
+  }
+
+  @Test def extends2(): Unit = withShell() {
+    val data = mktmp("ob;;smith")
+    val schema = "xsd/extends-1.dfdl.xsd"
+    s"parse --validate schematron={{$schema}} -s {{$schema}} $data" -> contains("</name>\n")
+  }
+
+  @Test def extends3(): Unit = withShell(FailureErrorCodeResult) {
+    val data = mktmp(";;smith")
+    val schema = "xsd/extends-1.dfdl.xsd"
+    s"parse --validate schematron={{$schema}} -s {{$schema}} $data" -> contains("</name>\n")
+  }
+
+  @Test def extends4(): Unit = withShell(FailureErrorCodeResult) {
+    val data = mktmp("bob;l;")
+    val schema = "xsd/extends-1.dfdl.xsd"
+    s"parse --validate schematron={{$schema}} -s {{$schema}} $data" -> contains("</name>\n")
+  }
+
+  @Test def extends5(): Unit = withShell(FailureErrorCodeResult) {
+    val data = mktmp(";l;")
+    val schema = "xsd/extends-1.dfdl.xsd"
+    s"parse --validate schematron={{$schema}} -s {{$schema}} $data" -> contains("</name>\n")
+  }
+
+  @Test def testWithNs1(): Unit = withShell() {
+    val data = mktmp("0;1")
+    val schema = "xsd/with-ns-1.dfdl.xsd"
+    s"parse --validate schematron={{$schema}} -s {{$schema}} $data" -> contains("</myns:interval>\n")
+  }
+
+  @Test def testWithNs2(): Unit = withShell(FailureErrorCodeResult) {
+    val data = mktmp("2;1")
+    val schema = "xsd/with-ns-1.dfdl.xsd"
+    s"parse --validate schematron={{$schema}} -s {{$schema}} $data" -> contains("</myns:interval>\n")
+  }
+
+  @Test def testWithNs3(): Unit = withShell(FailureErrorCodeResult) {
+    val data = mktmp("0;0")
+    val schema = "xsd/with-ns-1.dfdl.xsd"
+    s"parse --validate schematron={{$schema}} -s {{$schema}} $data" -> contains("</myns:interval>\n")
+  }
+}
```

Comment:
Lot's of \n's in this file. Likely going to cause issues on windows.

Reply:
Roger.  Will clean this up.